core/forms.py

```diff
@@ -14,6 +21,87 @@ def clean(self, values):
         return super(BetterReCaptchaField, self).clean(values)
 
 
+class AddOrganizerForm(forms.Form):
+    """
+    Custom form for adding new organizers to an existing event. 
+    
+    If user of given email already exists, they're added to the event and 
+    receive e-mail notification about it. 
+    
+    If user is new, they're created (randomly generated password), invited
+    to Slack and receive e-mail notification with instructions to login 
+    (including password).
+    """
+    event = forms.ModelChoiceField(queryset=Event.objects.all())
+    name = forms.CharField(label="Organizer's full name")
+    email = forms.CharField(label="E-mail address")
+
+    def __init__(self, event_choices=None, *args, **kwargs):
+        super(AddOrganizerForm, self).__init__(*args, **kwargs)
+        if event_choices:
+            self.fields['event'].queryset = event_choices
+
+    def generate_password(self):
```

Comment:
You can use [make_random_password](https://docs.djangoproject.com/en/1.9/topics/auth/customizing/#django.contrib.auth.models.BaseUserManager.make_random_password) instead 🙂